src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -326,9 +326,6 @@ def add_electrodes(
     else:
         channel_name_array = channel_ids.astype("str", copy=False)
         data_to_add["channel_name"].update(description="unique channel reference", data=channel_name_array, index=False)
-        # If the channel ids are integer keep the old behavior of asigning nwbfile.electrodes.id equal to channel_ids
-        if np.issubdtype(channel_ids.dtype, np.integer):
-            data_to_add["id"].update(data=channel_ids, index=False)
```

Comment:
Keeping integers `channel_ids` as `id` is incompatible with new changes

Reply:
Why is that? Couldn't you just cast the `channel_name` or `unit_name`  from an `int` to a string or something (I thought it already did this)? The table IDs must be unsigned integers, yes, but it can simply be a running count if needed, it's just for unique identification/indexing of rows

This seems to be the main culprit of the failing tests

Reply:
Imagine we have two probes with the same `int` channel ids, but belonging to different electrode groups. In that case, we want to be able to add electrodes from both probes, since they are not the same elelectrodes, even if they have the same `int` channel ids.

Reply:
Yes, but what I'm saying is there's a disconnect for just this purpose between the `id` of the table (just a unique integer identifier) and the `channel_name`. I like the logic of comparing the `channel_name` together with `group_name` to disambiguate - what I don't see is how a string casting of that `int`-valued `channel_id` on that group to serve as the `channel_name` causes an issue with that.

In fact you might be able to bypass any direct logic for this by simply using `enforce_unique_id=True` from https://hdmf.readthedocs.io/en/stable/tutorials/plot_dynamictable_howto.html#adding-rows

Reply:
Casting the `int`-channel_id to `str` doesn't cause the problem, but rather forcing the table `id` without using `enforce_unique_id=False`. Should we use `enforce_unique_id=False` when adding a channel/unit? IMO, since `channel_name` will already be a column, there is no point of forcing the `id` to be the `channel_id`, if `int`. But I'm happy to discuss!

Reply:
> but rather forcing the table id without using enforce_unique_id=False. Should we use enforce_unique_id=False when adding a channel/unit?

See the [Best Practice for Unique IDs](https://nwbinspector.readthedocs.io/en/dev/best_practices/tables.html#unique-ids). We should never use `enforce_unique_id=False` - what I'm saying is instead of specifying the `id` column manually, if you set `enforce_unique_id=True` it will automatically generate a unique `id` for each row.

What I'm saying is, we would treat the pair of `channel_name` (wherever it comes from) and `group_name` as the effective human-readable identifier of electrodes added via a recording extractor; but we also need the required ID field, too, but that doesn't have to match or correlate in any way to the pair of human-readable values.

Reply:
ok, so if I understand correctly, we would set the `channel_name` column values as `f{group_name}-{channel_id}`? Ok for me :) 

Sorry for the misunderstanding!

Reply:
> ok, so if I understand correctly, we would set the channel_name column values as f{group_name}-{channel_id}?

That sounds as if it could work! I was imagining `channel_name=str(channel_id)` as one column and `group_name=str(electrode_group.name)` as the other (on the same row), which would make the `channel_name` column have non-unique elements, but unique pairs taken across the two `_name` columns - but just adding extra info to the `channel_name` might be more convenient

Reply:
Actually, we already have the `channel_name` and `group_name` column and that's what is used to check if an electrode is already in the table: https://github.com/catalystneuro/neuroconv/blob/multi-probe/src/neuroconv/tools/spikeinterface/spikeinterface.py#L412-L433

I think this is a better approach, because we get to keep the original `channel_names`, which could come directly from the device (e.g. AP1... for spikeglx).

Maybe let me try to clarify: the behavior which was supported and in this new implementation should not be supported anymore is the following: 
```
# If the channel ids are integer keep the old behavior of asigning nwbfile.electrodes.id equal to channel_ids
if np.issubdtype(channel_ids.dtype, np.integer):
    data_to_add["id"].update(data=channel_ids, index=False)
```
Since now `channel_ids` can be non-unique in the electrodes table (if they belong to different groups), if we force the `id` to be the `channel_id` in case these are integers, we might have duplicate `id`, which of course we don't want to have. 
Basically, we have to move on to a new behavior :)

Reply:
I'm aware of all of this; but `id` is a required field of all tables, and its values must be unique. The `channel_ids`, for some time now, have been mapped instead for all practical purposes to `channel_name`. I'm saying, instead of mapping `channel_ids` to the `id` column at all, just use `enforce_unique_id=True` flag, don't include the `id` key when writing a row (i.e., in `data_to_add`), and it should take care of it automatically.

Let me know when the tests are passing and I can take a look at how you fixed it and go from there